lib/MusicBrainz/Server/Report/SeriesContainingDuplicates.pm

```diff
@@ -0,0 +1,95 @@
+package MusicBrainz::Server::Report::SeriesContainingDuplicates;
+use Moose;
+
+use MusicBrainz::Server::Constants qw( %PART_OF_SERIES );
+use MusicBrainz::Server::Data::Utils qw( type_to_model );
+use MusicBrainz::Server::Entity::Util::JSON qw( to_json_object );
+
+with 'MusicBrainz::Server::Report::SeriesReport',
+     'MusicBrainz::Server::Report::FilterForEditor::SeriesID';
+
+around inflate_rows => sub {
+    my $orig = shift;
+    my $self = shift;
+
+    my $items = $self->$orig(@_);
+
+    for my $result (@$items) {
+        my $series = $result->{series};
+        my $series_type = $self->c->model('SeriesType')->get_by_id($series->{typeID});
+        my $entity_type = $series_type->item_entity_type;
+        my $model = $self->c->model(type_to_model($entity_type));
+        my $entity = $model->get_by_id($result->{entity_id});
+        $result->{entity} = to_json_object($entity);
+    }
+
+    return $items;
+};
+
+sub query {
+    my $self = shift;
+
+    my @subqueries;
+
+    for my $entity_type (keys %PART_OF_SERIES) {
+        my $part_of_series_gid = $PART_OF_SERIES{$entity_type};
+        my ($series_table_data) = $self->c->model('Relationship')->generate_table_list(
+            'series',
+            $entity_type,
+        );
+        my ($table_name, $series_column, $other_column) = @$series_table_data;
+
+        my $subquery = <<~"SQL";
+            SELECT
+                DISTINCT s.id AS series_id,
+                         rels_table.$other_column AS entity_id,
+                         latv.text_value as order_number,
+                         s.name AS series_name
+            FROM
+                series s
+                JOIN $table_name rels_table ON rels_table.$series_column = s.id
+                JOIN link ON link.id = rels_table.link
+                JOIN link_type lt ON lt.id = link.link_type
+                LEFT JOIN link_attribute_text_value latv ON latv.link = link.id
+            WHERE (
+                lt.gid = \'$part_of_series_gid\' -- part of series
```

Comment:
Is there no way to pass query parameters here? If not, let's quote all injected values, just to be safe:
```Perl
my $part_of_series_gid = $self->c->sql->dbh->quote($PART_OF_SERIES{$entity_type});
```
```suggestion
                lt.gid = $part_of_series_gid -- part of series
```